src/ccip/CCIPLocalSimulatorFork.sol

```diff
@@ -112,124 +111,38 @@ contract CCIPLocalSimulatorFork is Test {
-                    s_processedMessages[message.messageId] = true;
-                    break;
-                }
-            }
-        }
+        uint256[] memory forkIds = new uint256[](1);
+        forkIds[0] = forkId;
 
-        vm.selectFork(forkId);
-        assertEq(vm.activeFork(), forkId);
-
-        IRouterFork.OffRamp[] memory offRamps =
-            IRouterFork(i_register.getNetworkDetails(block.chainid).routerAddress).getOffRamps();
-        length = offRamps.length;
-
-        for (uint256 i = length; i > 0; --i) {
-            if (offRamps[i - 1].sourceChainSelector == message.sourceChainSelector) {
-                vm.startPrank(offRamps[i - 1].offRamp);
-                uint256 numberOfTokens = message.tokenAmounts.length;
-                bytes[] memory offchainTokenData = new bytes[](numberOfTokens);
-                uint32[] memory tokenGasOverrides = new uint32[](numberOfTokens);
-                for (uint256 j; j < numberOfTokens; ++j) {
-                    tokenGasOverrides[j] = uint32(message.gasLimit);
-                }
-                IEVM2EVMOffRampPreV1dot6Fork(offRamps[i - 1].offRamp).executeSingleMessage(
-                    message, offchainTokenData, tokenGasOverrides
-                );
-                vm.stopPrank();
-                break;
-            }
-        }
+        _routeCapturedMessages(forkIds, sourceForkId, sourceRouterAddress);
     }
 
-    function _routePostV1dot6Message(uint256 forkId) internal {
-        Internal.EVM2AnyRampMessage memory message;
-        Vm.Log[] memory entries = vm.getRecordedLogs();
-        uint256 length = entries.length;
-        for (uint256 i; i < length; ++i) {
-            console2.logBytes32(entries[i].topics[0]);
-            if (entries[i].topics[0] == CCIPMessageSent.selector) {
-                message = abi.decode(entries[i].data, (Internal.EVM2AnyRampMessage));
-                if (!s_processedMessages[message.header.messageId]) {
-                    s_processedMessages[message.header.messageId] = true;
-                    break;
-                }
-            }
-        }
+    /**
+     * @notice  To be called after the sending of the cross-chain message (`ccipSend`).
+     *          Override variant of the `switchChainAndRouteMessage(uint256 forkId)` function in case of multiple destination forks.
+     *          Goes through the list of past logs and looks for the `CCIPSendRequested` and `CCIPMessageSent` events.
+     *          Loops through provided `forkIds` and tries to route the message to correct destination.
+     *          If you haven't provide correct `forkId`, the message will get lost.
+     *          If you sent more than one message, it will try to route all of them to correct destinations.
+     *
+     * @param forkIds - The IDs of the destination network forks. These are the returned values of `createFork()` or `createSelectFork()`
+     */
+    function switchChainAndRouteMessage(uint256[] memory forkIds) external {
```

Comment:
Rather than overload is it not better to have switchChainAndRouteMessages? Is there a specific advantage to overloading? since the two functions need to be separately implemented anyway, would it not be better to disambiguate their names?